pkg/disk/disk.go

```diff
@@ -119,39 +119,33 @@ func NewFSType(s string) (FSType, error) {
 	}
 }
 
-// PartitionTableType is the partition table type enum.
-type PartitionTableType uint64
+// PartitionTableType is the partition table type
+type PartitionTableType string
```

Comment:
I'd keep it being an enum and implement a custom JSON (un)marshaler. Such as in https://github.com/osbuild/images/blob/863bb93228db5e6e289f2d972e7af0e3e07eec69/pkg/sbom/document.go#L8-L48

Reply:
That is a good suggestion - I guess the advantage is that we get "free" error checking on unmarshal this way, something that just moving to a string would not give us. I like that.

Reply:
Updated now.